lib/index.js

```diff
@@ -17,40 +30,597 @@ const PERMISSIONS = bedrock.config.permission.permissions;
-        // as a prefix; a separate non-unique index is used for lookups
-        collection: 'account',
-        fields: {id: 1, email: 1},
-        options: {
-          partialFilterExpression: {email: {$exists: true}},
-          unique: true,
-          background: false
-        }
-      }, {
-        collection: 'account',
-        fields: {email: 1},
-        options: {
-          partialFilterExpression: {email: {$exists: true}},
-          unique: false,
-          background: false
-        }
-      }], callback);
-    }]
-  }, function(err) {
-    callback(err);
+const logger = bedrock.loggers.get('app').child('bedrock-account');
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await promisify(database.openCollections)(['account', 'identity']);
+
+  await promisify(database.createIndexes)([{
+    collection: 'account',
+    fields: {id: 1},
+    options: {unique: true, background: false}
+  }, {
+    // `id` is a prefix to allow for sharding on `id` -- a collection
+    // cannot be sharded unless its unique indexes have the shard key
+    // as a prefix; a separate non-unique index is used for lookups
+    collection: 'account',
+    fields: {id: 1, 'account.email': 1},
+    options: {
+      partialFilterExpression: {'account.email': {$exists: true}},
+      unique: true,
+      background: false
+    }
+  }, {
+    collection: 'account',
+    fields: {'account.email': 1},
+    options: {
+      partialFilterExpression: {'account.email': {$exists: true}},
+      unique: false,
+      background: false
+    }
+  }, {
+    // Note: Add index for identity management
+    // `id` is a prefix to allow for sharding on `id` -- a collection
+    // cannot be sharded unless its unique indexes have the shard key
+    // as a prefix; a separate non-unique index is used for lookups
+    collection: 'identity',
+    fields: {id: 1, 'meta.bedrock-account.account': 1},
+    options: {
+      partialFilterExpression: {
+        'meta.bedrock-account.account': {$exists: true}
+      },
+      unique: true,
+      background: false
+    }
+  }, {
+    // Note: Add index for identity management
+    collection: 'identity',
+    fields: {'meta.bedrock-account.account': 1},
+    options: {
+      partialFilterExpression: {
+        'meta.bedrock-account.account': {$exists: true}
+      },
+      unique: false,
+      background: false
+    }
+  }]);
+});
+
+/**
+ * Check for the existence of an account.
+ *
+ * @param actor the actor or capabilities for performing the action.
+ * @param [id] the ID of the account to check.
+ * @param [email] the email address for the account.
+ * @param [deleted] true to check accounts marked as deleted.
+ *
+ * @return a Promise that resolves to a boolean indicating account existence.
+ */
+api.exists = brCallbackify(async ({actor, id, email, deleted = false}) => {
+  assert.optionalString(id, 'id');
+  assert.optionalString(email, 'email');
+  assert.bool(deleted);
```

Comment:
I think the second argument (parameter name) is a good idea when using assert-plus as is the case with the previous assertions.  Here's a sample that shows with/without description argument: https://gist.github.com/mattcollier/00f8976169b9f5938128386756a7c68a

Reply:
Yeah, thanks. Just an oversight. We definitely want to use those.